### PR TITLE
Mobile: fix side-menu width on tablet devices

### PR DESCRIPTION
**Pull request by Tolu-Mals**
### Summary
This pull request fixes the issue of the side-menu becoming too wide on devices with wider screens like tablet devices. It's a part of the [Tablet Layout Project](https://discourse.joplinapp.org/t/joplin-mobile-tablet-layout/25775)

### CODE REVIEW

**Comment by laurent22** [comment]
Thanks Tolu. You have some conflicts that need to be resolved

**Comment by Tolu-Mals** [comment]
I've resolved the conflicts.

**Comment by laurent22** [comment]
There are CI errors. You need to run yarn install and commit again

**Comment by Tolu-Mals** [comment]
I'm not sure how to go about this. After running `yarn install`,  do I need to stage any changes or I'm fine with just running `git commit --allow-empty --allow-empty-message` and pushing?

File: packages/app-mobile/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+	preset: 'react-native',
+
+	testMatch: ['**/*.test.js'],
```
Comment (personalizedrefrigerator) [inline packages/app-mobile/jest.config.js]: It might make sense to use `ts-jest` for tests (I remember running into a type-related testing issue, but don't remember what it was... Perhaps it was because VSCode's Jest extension works best if the `.ts` files are used directly...)

See also:
 * https://reactnative.dev/blog/2018/05/07/using-typescript-with-react-native#adding-typescript-testing-infrastructure
 * https://github.com/laurent22/joplin/pull/6650/files#diff-c880a5ff48742e618045f29337622542edb0b94b6c8579c8b9df86a9e6bb02cb

File: packages/app-mobile/components/sideMenu.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	}
+}
+
+const SideMenu = connect((state: State) => {
+	return {
+		isOpen: state.showSideMenu,
+	};
+})(SideMenuComponent);
+
+export default SideMenu;
```
Comment (personalizedrefrigerator) [inline packages/app-mobile/components/sideMenu.ts]: Consider naming this file `SideMenu.ts` (to keep the name of the default export and the file name the same).

File: packages/app-mobile/components/getResponsiveValue.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
+// getResponsiveValue() returns the corresponding value for
+// of a particular device screen width based on a valueMap argument
+// and breakpoints
```
Comment (personalizedrefrigerator) [inline packages/app-mobile/components/getResponsiveValue.ts]:
```suggestion
// getResponsiveValue() returns the corresponding value for
// a particular device screen width based on a valueMap argument
// and breakpoints.
```

File: packages/app-mobile/components/getResponsiveValue.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+//
+// Breakpoints:
+// sm: < 481
+// md: >= 481
```
Comment (personalizedrefrigerator) [inline packages/app-mobile/components/getResponsiveValue.ts]:
```suggestion
// sm: < 481px
// md: >= 481px
```

File: packages/app-mobile/components/getResponsiveValue.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+//
+// More Info: https://discourse.joplinapp.org/t/week-4-report/26117
+import { Dimensions } from 'react-native';
+
+export interface ValueMap {
+    sm?: number;
+    md?: number;
+    lg?: number;
+    xl?: number;
+    xxl?: number;
```
Comment (personalizedrefrigerator) [inline packages/app-mobile/components/getResponsiveValue.ts]:
```suggestion
	/** Value to use on small-width displays */    
	sm?: number;
	/** Value to use on medium-width displays */
	md?: number;
	/** Value to use on large-width displays */
	lg?: number;
	/** Value to use on extra-large displays */
	xl?: number;
	/** Value to use on extra-extra-large displays */
	xxl?: number;
```

I think that documentation comments could be helpful here (I'm suggesting documentation comments because they have good support in `vscode`)!

Additionally, be sure not to mix tabs and spaces!

File: packages/app-mobile/components/getResponsiveValue.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+	if (md) {
+		value = md;
+	}
+
+	if (sm) {
+		value = sm;
+	}
+
+	if (width >= 481 && !!md) {
```
Comment (personalizedrefrigerator) [inline packages/app-mobile/components/getResponsiveValue.ts]:
```suggestion
	if (width >= 481 && md != null) {
```

If I run this:
```
$ node
Welcome to Node.js v16.16.0.
Type ".help" for more information.
> !!0
false
> !!1
true
```
As such, if someone sets `md` (or `lg`, `xl`, or `xxl`) to `0`, this conditional won't trigger.

(I'm suggesting `!=` and not `!==` because `md != null` returns `false` if `md` is `undefined`, but `md !== null` returns `true` if `md` is `undefined`).

File: packages/app-mobile/root.tsx [hunk 7]
```diff
@@ -682,8 +683,21 @@ class AppComponent extends React.Component {
...
+			const sideMenuWidth = getResponsiveValue({
+				sm: 250,
+				md: 260,
+				lg: 270,
+				xl: 280,
+				xxl: 290,
+			});
+
+			return sideMenuWidth;
+		};
```
Comment (personalizedrefrigerator) [inline packages/app-mobile/root.tsx]: It might make sense to make this a `private` method.

**Review by personalizedrefrigerator — COMMENTED** [review]
Nice work so far! Here's what I've noticed from a quick look through the code.

File: packages/app-mobile/jest.config.js (hunk 1 above)
Comment (Tolu-Mals) [reply packages/app-mobile/jest.config.js]: Thanks for this. I'll look through it.

File: packages/app-mobile/components/sideMenu.ts (hunk 2 above)
Comment (Tolu-Mals) [reply packages/app-mobile/components/sideMenu.ts]: Noted, thanks

File: packages/app-mobile/components/getResponsiveValue.ts (hunk 5 above)
Comment (Tolu-Mals) [reply packages/app-mobile/components/getResponsiveValue.ts]: Thanks for this.

File: packages/app-mobile/components/getResponsiveValue.ts (hunk 6 above)
Comment (Tolu-Mals) [reply packages/app-mobile/components/getResponsiveValue.ts]: Wasn't sure initially if getResponsiveValue should accept a value of zero, but I think it makes sense that it does. Thank you.

File: packages/app-mobile/jest.config.js (hunk 1 above)
Comment (Tolu-Mals) [reply packages/app-mobile/jest.config.js]: @laurent22 Is it okay to use Typescript directly for testing?

Comment (laurent22) [reply packages/app-mobile/jest.config.js]: Eventually we might do that, but for now let's keep using the compiled .js files for testing.

File: packages/app-mobile/components/getResponsiveValue.ts (hunk 5 above)
Comment (laurent22) [reply packages/app-mobile/components/getResponsiveValue.ts]: Good point about the comments but should be single lines one `//`. Please only use that type of comments

File: packages/app-mobile/components/getResponsiveValue.ts [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    sm?: number;
+	/** Value to use on medium-width displays */
+    md?: number;
+	/** Value to use on large-width displays */
+    lg?: number;
+	/** Value to use on extra-large width displays */
+    xl?: number;
+	/** Value to use on extra-extra-large width displays */
+    xxl?: number;
+}
```
Comment (laurent22) [inline packages/app-mobile/components/getResponsiveValue.ts]: You're mixing tabs and spaces here. Please make sure you run `yarn run linter ./` from the root of the repository before committing.

**Comment by laurent22** [comment]
Please resolve the conflicts.

**Comment by laurent22** [comment]
Thanks for the review @personalizedrefrigerator! @Tolu-Mals, nearly there - just these small issues I mentioned and we can merge.

File: packages/app-mobile/components/getResponsiveValue.ts (hunk 5 above)
Comment (Tolu-Mals) [reply packages/app-mobile/components/getResponsiveValue.ts]: Alright, noted.

File: packages/app-mobile/components/getResponsiveValue.ts (hunk 8 above)
Comment (Tolu-Mals) [reply packages/app-mobile/components/getResponsiveValue.ts]: Noted, I'll fix that.

**Comment by personalizedrefrigerator** [comment]
It looks like the mobile tests aren't being found:
```
➤YN0000: [@joplin/app-mobile]: Process started
➤YN0000: [@joplin/app-mobile]: No tests found, exiting with code 1
➤YN0000: [@joplin/app-mobile]: Run with `--passWithNoTests` to exit with code 0
➤YN0000: [@joplin/app-mobile]: In /home/runner/work/joplin/joplin/packages/app-mobile
➤YN0000: [@joplin/app-mobile]:   175 files checked.
➤YN0000: [@joplin/app-mobile]:   testMatch: **/*.test.js - 0 matches
➤YN0000: [@joplin/app-mobile]:   testPathIgnorePatterns: /home/runner/work/joplin/joplin/packages/app-mobile/node_modules/ - 175 matches
➤YN0000: [@joplin/app-mobile]:   testRegex:  - 0 matches
➤YN0000: [@joplin/app-mobile]: Pattern:  - 0 matches
➤YN0000: [@joplin/app-mobile]: Process exited (exit code 1), completed in 4s 175ms
```
https://github.com/laurent22/joplin/runs/7471464033?check_suite_focus=true#step:10:4272

You might need to update `tsconfig.json` — my earlier PR configured things to use `.test.ts` files directly (instead of compiling them into `.test.js` files).

**Comment by Tolu-Mals** [comment]
> It looks like the mobile tests aren't being found:
> 
> ```
> ➤YN0000: [@joplin/app-mobile]: Process started
> ➤YN0000: [@joplin/app-mobile]: No tests found, exiting with code 1
> ➤YN0000: [@joplin/app-mobile]: Run with `--passWithNoTests` to exit with code 0
> ➤YN0000: [@joplin/app-mobile]: In /home/runner/work/joplin/joplin/packages/app-mobile
> ➤YN0000: [@joplin/app-mobile]:   175 files checked.
> ➤YN0000: [@joplin/app-mobile]:   testMatch: **/*.test.js - 0 matches
> ➤YN0000: [@joplin/app-mobile]:   testPathIgnorePatterns: /home/runner/work/joplin/joplin/packages/app-mobile/node_modules/ - 175 matches
> ➤YN0000: [@joplin/app-mobile]:   testRegex:  - 0 matches
> ➤YN0000: [@joplin/app-mobile]: Pattern:  - 0 matches
> ➤YN0000: [@joplin/app-mobile]: Process exited (exit code 1), completed in 4s 175ms
> ```
> 
> https://github.com/laurent22/joplin/runs/7471464033?check_suite_focus=true#step:10:4272
> 
> You might need to update `tsconfig.json` — my earlier PR configured things to use `.test.ts` files directly (instead of compiling them into `.test.js` files).

Thanks. I used my own tsconfig while resolving the merge conflicts. I'm working on it though.

**Comment by Tolu-Mals** [comment]
> Thanks for the review @personalizedrefrigerator! @Tolu-Mals, nearly there - just these small issues I mentioned and we can merge.

Whenever I run `yarn run tsc` the typescript test files aren't getting compiled even locally.

**Comment by Tolu-Mals** [comment]
> > It looks like the mobile tests aren't being found:
> > ```
> > ➤YN0000: [@joplin/app-mobile]: Process started
> > ➤YN0000: [@joplin/app-mobile]: No tests found, exiting with code 1
> > ➤YN0000: [@joplin/app-mobile]: Run with `--passWithNoTests` to exit with code 0
> > ➤YN0000: [@joplin/app-mobile]: In /home/runner/work/joplin/joplin/packages/app-mobile
> > ➤YN0000: [@joplin/app-mobile]:   175 files checked.
> > ➤YN0000: [@joplin/app-mobile]:   testMatch: **/*.test.js - 0 matches
> > ➤YN0000: [@joplin/app-mobile]:   testPathIgnorePatterns: /home/runner/work/joplin/joplin/packages/app-mobile/node_modules/ - 175 matches
> > ➤YN0000: [@joplin/app-mobile]:   testRegex:  - 0 matches
> > ➤YN0000: [@joplin/app-mobile]: Pattern:  - 0 matches
> > ➤YN0000: [@joplin/app-mobile]: Process exited (exit code 1), completed in 4s 175ms
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > https://github.com/laurent22/joplin/runs/7471464033?check_suite_focus=true#step:10:4272
> > You might need to update `tsconfig.json` — my earlier PR configured things to use `.test.ts` files directly (instead of compiling them into `.test.js` files).
> 
> Thanks. I used my own tsconfig while resolving the merge conflicts. I'm working on it though.

**Comment by Tolu-Mals** [comment]
> It looks like the mobile tests aren't being found:
> 
> ```
> ➤YN0000: [@joplin/app-mobile]: Process started
> ➤YN0000: [@joplin/app-mobile]: No tests found, exiting with code 1
> ➤YN0000: [@joplin/app-mobile]: Run with `--passWithNoTests` to exit with code 0
> ➤YN0000: [@joplin/app-mobile]: In /home/runner/work/joplin/joplin/packages/app-mobile
> ➤YN0000: [@joplin/app-mobile]:   175 files checked.
> ➤YN0000: [@joplin/app-mobile]:   testMatch: **/*.test.js - 0 matches
> ➤YN0000: [@joplin/app-mobile]:   testPathIgnorePatterns: /home/runner/work/joplin/joplin/packages/app-mobile/node_modules/ - 175 matches
> ➤YN0000: [@joplin/app-mobile]:   testRegex:  - 0 matches
> ➤YN0000: [@joplin/app-mobile]: Pattern:  - 0 matches
> ➤YN0000: [@joplin/app-mobile]: Process exited (exit code 1), completed in 4s 175ms
> ```
> 
> https://github.com/laurent22/joplin/runs/7471464033?check_suite_focus=true#step:10:4272
> 
> You might need to update `tsconfig.json` — my earlier PR configured things to use `.test.ts` files directly (instead of compiling them into `.test.js` files).

The typescript files for the tests aren't being compiled when I run 'yarn run tsc' locally.

**Comment by Tolu-Mals** [comment]
> It looks like the mobile tests aren't being found:
> 
> ```
> ➤YN0000: [@joplin/app-mobile]: Process started
> ➤YN0000: [@joplin/app-mobile]: No tests found, exiting with code 1
> ➤YN0000: [@joplin/app-mobile]: Run with `--passWithNoTests` to exit with code 0
> ➤YN0000: [@joplin/app-mobile]: In /home/runner/work/joplin/joplin/packages/app-mobile
> ➤YN0000: [@joplin/app-mobile]:   175 files checked.
> ➤YN0000: [@joplin/app-mobile]:   testMatch: **/*.test.js - 0 matches
> ➤YN0000: [@joplin/app-mobile]:   testPathIgnorePatterns: /home/runner/work/joplin/joplin/packages/app-mobile/node_modules/ - 175 matches
> ➤YN0000: [@joplin/app-mobile]:   testRegex:  - 0 matches
> ➤YN0000: [@joplin/app-mobile]: Pattern:  - 0 matches
> ➤YN0000: [@joplin/app-mobile]: Process exited (exit code 1), completed in 4s 175ms
> ```
> 
> https://github.com/laurent22/joplin/runs/7471464033?check_suite_focus=true#step:10:4272
> 
> You might need to update `tsconfig.json` — my earlier PR configured things to use `.test.ts` files directly (instead of compiling them into `.test.js` files).

Oh it's exactly as you said. @personalizedrefrigerator . I read 'jest config' when you mentioned 'tsconfig.json' earlier. The current tsconfig.json is ignoring typescript test files. Fixing that now.

**Comment by personalizedrefrigerator** [comment]
It looks like a dependency is missing! Try running `yarn install` again and committing changes to `yarn.lock`.

>  Error: ➤YN0000: │ root@workspace:.STDERR➤YN0000: [@joplin/app-mobile]: tools/buildInjectedJs.ts(6,45): error TS7016: Could not find a declaration file for module 'fs-extra'. 'D:/a/joplin/joplin/packages/app-mobile/node_modules/fs-extra/lib/index.js' implicitly has an 'any' type.

(A dependency on `@types/fs-extra` was added in a recently-merged PR)

File: packages/app-mobile/jest.config.js (hunk 1 above)
Comment (laurent22) [reply packages/app-mobile/jest.config.js]: Wait, actually @personalizedrefrigerator set it up with ts-jest, is that right? In that case, yes please use this for this particular package.

**Comment by Tolu-Mals** [comment]
> It looks like a dependency is missing! Try running `yarn install` again and committing changes to `yarn.lock`.
> 
> > Error: ➤YN0000: │ root@workspace:.STDERR➤YN0000: [@joplin/app-mobile]: tools/buildInjectedJs.ts(6,45): error TS7016: Could not find a declaration file for module 'fs-extra'. 'D:/a/joplin/joplin/packages/app-mobile/node_modules/fs-extra/lib/index.js' implicitly has an 'any' type.
> 
> (A dependency on `@types/fs-extra` was added in a recently-merged PR)

Thanks. I'll try that.

**Comment by Tolu-Mals** [comment]
> It looks like a dependency is missing! Try running `yarn install` again and committing changes to `yarn.lock`.
> 
> > Error: ➤YN0000: │ root@workspace:.STDERR➤YN0000: [@joplin/app-mobile]: tools/buildInjectedJs.ts(6,45): error TS7016: Could not find a declaration file for module 'fs-extra'. 'D:/a/joplin/joplin/packages/app-mobile/node_modules/fs-extra/lib/index.js' implicitly has an 'any' type.
> 
> (A dependency on `@types/fs-extra` was added in a recently-merged PR)

After running `yarn install` there were no changes in `yarn.lock`

**Comment by Tolu-Mals** [comment]
A test is now failing due to a linter error in `getResponsiveValue.ts`:

```
  41:10  error  Expected '!==' and instead saw '!='  eqeqeq
  45:9   error  Expected '!==' and instead saw '!='  eqeqeq
  49:9   error  Expected '!==' and instead saw '!='  eqeqeq
  57:9   error  Expected '!==' and instead saw '!='  eqeqeq
  65:25  error  Expected '!==' and instead saw '!='  eqeqeq
  69:26  error  Expected '!==' and instead saw '!='  eqeqeq
  73:27  error  Expected '!==' and instead saw '!='  eqeqeq
```

eslint expects only strict equality to be used. I'll see if there's a way to rewrite the logic without the '!='

**Comment by Tolu-Mals** [comment]
I think this is ready to be merged.

File: packages/app-mobile/components/getResponsiveValue.ts [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+		value = lg;
+	}
+
+	if ((!!md) || (md === 0)) {
+		value = md;
+	}
+
+	if ((!!sm) || (sm === 0)) {
+		value = sm;
+	}
```
Comment (personalizedrefrigerator) [inline packages/app-mobile/components/getResponsiveValue.ts]: This can be simplified using the [`??` operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator):
```suggestion
	value = sm ?? md ?? lg ?? xl ?? xxl;
```

File: packages/app-mobile/components/getResponsiveValue.ts [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+		value = lg;
+	}
+
+	if (width >= 1025 && ((!!xl) || (xl === 0))) {
+		value = xl;
+	}
+
+	if (width >= 1201 && ((!!xxl) || (xxl === 0))) {
+		value = xxl;
+	}
```
Comment (personalizedrefrigerator) [inline packages/app-mobile/components/getResponsiveValue.ts]: This might be simpler:
```suggestion
	if (width >= 481) {
		value = md ?? value;
	}

	if (width >= 769) {
		value = lg ?? value;
	}

	if (width >= 1025) {
		value = xl ?? value;
	}

	if (width >= 1201) {
		value = xxl ?? value;
	}
```
It might also be more difficult to read. I'm not sure which I like better...

File: packages/app-mobile/components/getResponsiveValue.ts (hunk 9 above)
Comment (Tolu-Mals) [reply packages/app-mobile/components/getResponsiveValue.ts]: Awesome, that's a lot cleaner. Thank you.

File: packages/app-mobile/components/getResponsiveValue.ts (hunk 10 above)
Comment (Tolu-Mals) [reply packages/app-mobile/components/getResponsiveValue.ts]: This makes more sense in my opinion. Should not be too difficult to read if they understand what the `??` operator does. Thanks again!

File: packages/app-mobile/jest.config.js (hunk 1 above)
Comment (Tolu-Mals) [reply packages/app-mobile/jest.config.js]: I'm just seeing this now.

File: packages/app-mobile/jest.config.js [hunk 11]
```diff
@@ -1,17 +1,9 @@
-// Test configuration
-// See https://jestjs.io/docs/configuration#testenvironment-string
+module.exports = {
+	preset: 'react-native',
 
-const config = {
-	preset: 'ts-jest',
+	testMatch: ['**/*.test.js'],
```
Comment (laurent22) [inline packages/app-mobile/jest.config.js]: Why was ts-test removed? I thought we were keeping this?

File: .gitignore [hunk 12]
```diff
@@ -2128,3 +2140,6 @@ packages/tools/website/utils/types.d.ts
 packages/tools/website/utils/types.js
 packages/tools/website/utils/types.js.map
 # AUTO-GENERATED - EXCLUDED TYPESCRIPT BUILD
+packages/app-mobile/components/get-responsive-value.test.js
+packages/app-mobile/components/get-responsive-value.test.js
+packages/app-mobile/components/get-responsive-value.test.js
```
Comment (laurent22) [inline .gitignore:2217]: I don't think that should be here - run yarn run updateIgnored from the root

File: packages/app-mobile/components/SideMenu.ts [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+import { State } from '@joplin/lib/reducer';
+
+type Event = {
+	nativeEvent: {
+		layout: {
+			width: number;
+			height: number;
+		};
+	};
+};
```
Comment (laurent22) [inline packages/app-mobile/components/SideMenu.ts]: Please remove. We can't have each file define types for third-party libraries. That should maybe be in a separate file shared by all other files. But for now just remove.

File: packages/app-mobile/tsconfig.json [hunk 14]
```diff
@@ -6,11 +6,5 @@
 	],
 	"exclude": [
 		"**/node_modules",
-
-		// Files that don't need transpilation
-		"gulpfile.ts",
-		"tools/*.ts",
-		"**/*.test.ts",
-		"**/*.test.tsx",
```
Comment (laurent22) [inline packages/app-mobile/tsconfig.json]: Same here, why was that changed?

File: packages/app-mobile/jest.config.js (hunk 11 above)
Comment (Tolu-Mals) [reply packages/app-mobile/jest.config.js]: > Wait, actually @personalizedrefrigerator set it up with ts-jest, is that right? In that case, yes please use this for this particular package.

I saw this a bit late. I'll fix that.

File: .gitignore (hunk 12 above)
Comment (Tolu-Mals) [reply .gitignore:2217]: Noted.

File: packages/app-mobile/components/SideMenu.ts (hunk 13 above)
Comment (Tolu-Mals) [reply packages/app-mobile/components/SideMenu.ts]: Noted.

File: packages/app-mobile/tsconfig.json (hunk 14 above)
Comment (Tolu-Mals) [reply packages/app-mobile/tsconfig.json]: I'll fix that.

File: packages/app-mobile/jest.config.js (hunk 11 above)
Comment (Tolu-Mals) [reply packages/app-mobile/jest.config.js]: I'm currently setting up app-mobile to use ts-jest for testing and I've realized that the preset doesn't need to be 'ts-jest' for ts-jest to be used. The preset needs to 'react-native'. Just leaving this here to avoid any confusion. I followed this guide: [Using Typescript With React Native](https://reactnative.dev/blog/2018/05/07/using-typescript-with-react-native#adding-typescript-testing-infrastructure).

File: .gitignore (hunk 12 above)
Comment (Tolu-Mals) [reply .gitignore:2217]: I don't know why but running `updateIgnored` from root doesn't remove it from the .gitignore

File: packages/app-mobile/jest.config.js (hunk 11 above)
Comment (laurent22) [reply packages/app-mobile/jest.config.js]: I don't know about this, but I think that's going to conflict with what @personalizedrefrigerator is doing? @personalizedrefrigerator will your tests still work with this change?

Comment (personalizedrefrigerator) [reply packages/app-mobile/jest.config.js]: Currently, `tsconfig.json` is configured to not create .js files from `.test.ts` files. As such, unless `tsconfig.json` is modified, it shouldn't work with my test files.

Comment (Tolu-Mals) [reply packages/app-mobile/jest.config.js]: I changed `tsconfig` in my recent [commit](https://github.com/laurent22/joplin/pull/6662/commits/45962c7a4b982cec093fb81f1a3486adcfa10bbb#diff-dcc2d9d32ba5701b440eec8abf05054bb74f35ad002fa7c2e1107e964858f8d0). Your tests are running fine now as well as mine.

Comment (laurent22) [reply packages/app-mobile/jest.config.js]: I would like some consistency in the tests. Either it's all .js or it's all .ts but not mix and match. I liked the approach of running the .test.ts files directly because it means we don't need source maps.

Is it not possible to get that working with your tests @Tolu-Mals? If there's an issue please explain clearly what it is.

Comment (Tolu-Mals) [reply packages/app-mobile/jest.config.js]: It's possible. I'll do that.

**Comment by laurent22** [comment]
I think it's good now but please resolve the yarn.lock conflict

**Comment by laurent22** [comment]
Tests are failing: https://github.com/laurent22/joplin/runs/7729267796?check_suite_focus=true#step:10:4324

**Comment by Tolu-Mals** [comment]
> Tests are failing: https://github.com/laurent22/joplin/runs/7729267796?check_suite_focus=true#step:10:4324

Thank you. I just made a fix, I think they should pass now.

**Comment by Tolu-Mals** [comment]
@laurent22 It's good now.

**Comment by laurent22** [comment]
Unfortunately still a yarn.lock conflict. A new one this time.

**Comment by laurent22** [comment]
Ok I've tested it on simulator and it looks good, thanks Tolu!
